libcloudforensics/providers/azure/internal/compute.py

```diff
@@ -36,47 +44,508 @@
+    self.az_account.storage.DeleteStorageAccount(storage_account_name)
+
+    return AZComputeDisk(self.az_account,
+                         disk.id,
+                         disk.name,
+                         disk.location,
+                         disk.zones)
+
+  def GetOrCreateAnalysisVm(
+      self,
+      vm_name: str,
+      boot_disk_size: int,
+      cpu_cores: int,
+      memory_in_mb: int,
+      ssh_public_key: str,
+      region: Optional[str] = None,
+      packages: Optional[List[str]] = None,
+      tags: Optional[Dict[str, str]] = None
+  ) -> Tuple['AZComputeVirtualMachine', bool]:
+    """Get or create a new virtual machine for analysis purposes.
+
+    Args:
+      vm_name (str): The instance name tag of the virtual machine.
+      boot_disk_size (int): The size of the analysis VM boot volume (in GB).
+      cpu_cores (int): Number of CPU cores for the analysis VM.
+      memory_in_mb (int): The memory size (in MB) for the analysis VM.
+      ssh_public_key (str): A SSH public key data to associate with the
+          VM. This must be provided as otherwise the VM will not be
+          accessible.
+      region (str): Optional. The region in which to create the vm. If not
+          provided, the vm will be created in the default_region
+          associated to the AZAccount object.
+      packages (List[str]): Optional. List of packages to install in the VM.
+      tags (Dict[str, str]): Optional. A dictionary of tags to add to the
+          instance, for example {'TicketID': 'xxx'}. An entry for the
+          instance name is added by default.
+
+    Returns:
+      Tuple[AZComputeVirtualMachine, bool]: A tuple with an
+          AZComputeVirtualMachine object and a boolean indicating if the
+          virtual machine was created (True) or reused (False).
+
+    Raises:
+      RuntimeError: If the virtual machine cannot be found or created.
+    """
+
+    # Re-use instance if it already exists, or create a new one.
+    try:
+      instance = self.GetInstance(vm_name)
+      if instance:
+        created = False
+        return instance, created
+    except RuntimeError:
+      pass
+
+    # Validate SSH public key format
+    try:
+      sshpubkeys.SSHKey(ssh_public_key, strict=True).parse()
+    except sshpubkeys.InvalidKeyError as exception:
+      raise RuntimeError('The provided public SSH key is invalid: '
+                         '{0:s}'.format(str(exception)))
+
+    instance_type = self._GetInstanceType(cpu_cores, memory_in_mb)
+    startup_script = utils.ReadStartupScript()
+    if packages:
+      startup_script = startup_script.replace('${packages[@]}', ' '.join(
+          packages))
+
+    if not region:
+      region = self.az_account.default_region
+
+    creation_data = {
+        'location': region,
+        'properties': {
+            'hardwareProfile': {'vmSize': instance_type},
+            'storageProfile': {
+                'imageReference': {
+                    'sku': '18.04-LTS',
```

Comment:
Is this something we want to put in the code rather than config?
And if so, do we want it buried deep within a data blob rather than a constant at the top?

Reply:
Moved to `common.py`